### PR TITLE
fix override result will be ignored by ReviseReplica issue

### PR DESCRIPTION
**Pull request by likakuli**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Replica set by overridepolicy in its Plaintext will be reset by ReviseReplica interperterhook

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/1347

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
fix a bug that Replica set by overridepolicy will be reset by ReviseReplica interpreterhook
```

### CODE REVIEW

File: pkg/controllers/binding/common.go [hunk 1]
```diff
@@ -116,6 +111,13 @@ func ensureWork(
 			}
 		}
 
+		// Call ReviseReplica func first, as ApplyOverridePolicy has the highest priority
```
Comment (iawia002) [inline pkg/controllers/binding/common.go:114]: This comment is a bit ambiguous, as it refers to the `ReviseReplica` function after it has already been called, I suggest adjusting it:

```suggestion
		// We should call ApplyOverridePolicies last, as override rules have the highest priority
```

Comment (likakuli) [reply pkg/controllers/binding/common.go]: updated

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Comment by RainbowMango** [comment]
/retitle fix override result will be ignored by ReviseReplica issue

**Pull request merged**